src/otx/core/model/base.py

```diff
@@ -389,6 +389,13 @@ def on_load_checkpoint(self, checkpoint: dict[str, Any]) -> None:
         super().on_load_checkpoint(checkpoint)
 
         if ckpt_label_info := checkpoint.get("label_info", None):
+            if isinstance(ckpt_label_info, LabelInfo) and not hasattr(ckpt_label_info, "label_ids"):
+                # NOTE: This is for backward compatibility
+                ckpt_label_info = LabelInfo(
+                    label_groups=ckpt_label_info.label_groups,
+                    label_names=ckpt_label_info.label_names,
+                    label_ids=[str(i) for i in range(len(ckpt_label_info.label_names))],
```

Comment:
I have a little doubt if we have to replicate `label_names` here, but let's look how it works in Geti.